scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/policy/edf/JobMostLikelyMissedEmailNotification.java

```diff
@@ -0,0 +1,151 @@
+ * Workflows & Scheduling, Orchestration, Cloud Automation
+ * and Big Data Analysis on Enterprise Grids & Clouds.
+ *
+ * Copyright (c) 2007 - 2017 ActiveEon
+ * Contact: contact@activeeon.com
+ *
+ * This library is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License
+ * as published by the Free Software Foundation: version 3 of
+ * the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive.scheduler.policy.edf;
+
+import static org.ow2.proactive.scheduler.core.JobEmailNotification.GENERIC_INFORMATION_KEY_EMAIL;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.text.SimpleDateFormat;
+import java.time.Duration;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.Objects;
+import java.util.TimeZone;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.log4j.Logger;
+import org.ow2.proactive.addons.email.exception.EmailException;
+import org.ow2.proactive.scheduler.core.JobEmailNotificationException;
+import org.ow2.proactive.scheduler.core.properties.PASchedulerProperties;
+import org.ow2.proactive.scheduler.job.InternalJob;
+import org.ow2.proactive.scheduler.util.JobLogger;
+import org.ow2.proactive.scheduler.util.SendMail;
+
+
+public class JobMostLikelyMissedEmailNotification {
+
+    private static final Logger logger = Logger.getLogger(JobMostLikelyMissedEmailNotification.class);
+
+    private static final JobLogger jlogger = JobLogger.getInstance();
+
+    private final InternalJob internalJob;
+
+    private final Date expectedFinishingTime;
+
+    private final Date deadline;
+
+    private Duration jobExecutionTime = null;
+
+    public JobMostLikelyMissedEmailNotification(InternalJob internalJob, Date expectedFinishingTime, Date deadline) {
+        this.internalJob = internalJob;
+        this.expectedFinishingTime = expectedFinishingTime;
+        this.deadline = deadline;
+    }
+
+    public JobMostLikelyMissedEmailNotification(InternalJob internalJob, Date expectedFinishingTime, Date deadline,
+            Duration jobExecutionTime) {
+        this.internalJob = internalJob;
+        this.expectedFinishingTime = expectedFinishingTime;
+        this.deadline = deadline;
+        this.jobExecutionTime = jobExecutionTime;
+    }
+
+    public static void main(String[] args) {
+        System.out.println(new Date().getTime());
+        System.out.println(System.currentTimeMillis());
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this use of System.out or System.err by a logger. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)